MapboxVision/Services/BatteryManager.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+
+/// Class that contains logic to dispatch and manage the battery state of the device.
+final class BatteryManager {
+    /// Constants that represents significant battery power states.
+    private enum ConstantsBatteryLevel {
+        /// The value that corresponds to fully discharged battery (the device might be still powered on).
+        static let fullyDischarged: Float = 0.0
+        /// The value that corresponds to situation when battery state is unknown or battery monitoring is not enabled.
+        static let unavailable: Int8 = -1
+    }
+
+    // MARK: Public properties
+
+    /// Indicates whether battery is currently charging or not.
+    private(set) var isCharging = false
+    /// Battery level in range from 0 (fully discharged) to 100 (100% charged). In case when battery level can't be collected the return value is -1.
+    private(set) var batteryLevel = ConstantsBatteryLevel.unavailable
+
+    // MARK: Lifecycle
+
+    /**
+     Base initializer to create an instance of `BatteryManager`.
+     */
+    init() {
+        UIDevice.current.isBatteryMonitoringEnabled = true
+        NotificationCenter.default.addObserver(self,
```

Comment:
Why do we need to subscribe to update events if we're just pulling a value which is accessible via accessors?

Reply:
Good question. This is because we don't need to pull the UIDevice's value each time (for now update interval in core equals to 60 seconds). We can use cached value from BatteryManager instead and update it only when the battery level/state is changed.

Reply:
Agreed to get rid of subscriptions.